str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+
+
+def compute_final_call(n_total_well_supported_motifs, n_pathogenic_motifs, n_benign_motifs):
+    """Decide which combination of motifs is supported by the data.
+
+    NOTE: there's no attempt to determine the size of the expansion and whether it's in the pathogenic range.
+
+    Return:
+        str: a string such as "BENIGN MOTIF / PATHOGENIC MOTIF" indicating which motifs were detected
+    """
+
+    if n_total_well_supported_motifs == 0:
+        final_call = NO_CALL1
+    elif n_pathogenic_motifs == n_total_well_supported_motifs:
+        # reads support only known pathogenic motif(s)
+        final_call = PATHOGENIC_PATHOGENIC_CALL
+    elif n_benign_motifs == n_total_well_supported_motifs:
+        # reads support only known benign motif(s)
+        final_call = BENIGN_BENIGN_CALL
+    elif n_benign_motifs == 0 and n_pathogenic_motifs == 0:
+        # reads support one or more non-reference motifs of unknown significance
+        final_call = UNCERTAIN_SIG_UNCERTAIN_SIG_CALL
+    elif n_benign_motifs > 0 and n_pathogenic_motifs > 0:
+        # reads support one known benign motif and one pathogenic motif
+        final_call = BENIGN_PATHOGENIC_CALL
+    elif n_pathogenic_motifs > 0:
+        # reads support one pathogenic motif and at least one other motif of unknown significance
+        final_call = PATHOGENIC_UNCERTAIN_SIG_CALL
+    elif n_benign_motifs > 0:
+        # reads support one known benign motif and at least one other motif of unknown significance
+        final_call = BENIGN_UNCERTAIN_SIG_CALL
+    else:
+        raise Exception(f"Unexpected state when n_total_well_supported_motifs={n_total_well_supported_motifs}, "
+                        f"n_pathogenic_motifs={n_pathogenic_motifs}, n_benign_motifs={n_benign_motifs}")
+
+    return final_call
+
+
+def process_offtarget_regions(motif, motif_number, read_count, flank_coverage_mean, args, locus_results_json):
+    """Look for off-target reads for a specific motif.
+
+    Args:
+        motif (str): Repeat unit
+        motif_number (int): Motif number relative to other selected motifs.
+        read_count (int): Not-offtarget read count.
+        flank_coverage_mean (float): mean coverage in flanking regions
+        args (object): command-line arguments from argparse.
+        locus_results_json (dict): Results will be added to this dictionary.
+    """
+    canonical_motif = compute_canonical_repeat_unit(motif)
+    offtarget_regions = OFFTARGET_REGIONS[args.genome_version].get(canonical_motif)
+    if offtarget_regions is None:
+        print(f"WARNING: off-target regions not available for {canonical_motif}")
+        return
+
+    read_count_with_offtargets = read_count
+    with pysam.Samfile(args.bam_or_cram_path, reference_filename=args.reference_fasta) as f:
+        for offtarget_region in offtarget_regions:
+            offtarget_chrom, offtarget_start, offtarget_end = parse_interval(offtarget_region)
```

Comment:
parse_interval is agnostic to 0- or 1-based coords. If offtarget_region is 0-based, it will return 0-based coords.
I don't specify 0- or 1-based coords in the off-target region var names (like do in other places) because off-target regions are inexact so it doesn't matter as much.